### PR TITLE
Improvements and IPv6 Support

### PR DESCRIPTION
**Pull request by joker234**
I did some Improvements on your litte script. (btw. thanks for that) I wrote some verbose commit messages where is described why I did what. If you want, I can do patches for each of this commits for you.

### CODE REVIEW

**Comment by gehaxelt** [comment]
Hi,
sorry for the short delay, but I got the email notification just some minutes ago. 

I'll have a look at your changes tomorrow and comment/merge it then. 

Thanks in advance for your work,
gehaxelt

File: dnsupdate.sh [hunk 1]
```diff
@@ -2,14 +2,25 @@
...
+OLDIPv6=$(cat old.ipv6)
+NEWIPv4=$(curl -s curlmyip.com)
+NEWIPv6=$(curl -s curlmyip6.com)
 
-if [ ! "$OLDIP" == "$NEWIP" ]; then
-    echo $NEWIP > old.ip
-    DATA=$(cat update.api | sed "s/%PASSWD%/$PASSWORD/g;s/%USER%/$USERNAME/g;s/%DNSID%/$DNSID/g;s/%NEWIP%/$NEWIP/g")
-    curl  -X POST -d "$DATA" "$APIHOST" --header "Content-Type:text/xml" --insecure > update.log
+if [ ! "$OLDIPv4" == "$NEWIPv4" ]; then
+    echo $NEWIP > old.ipv4
```
Comment (gehaxelt) [inline dnsupdate.sh]: I think this should be ```$NEWIPv4```, because ```$NEWIP``` isn't defined anymore.

File: dnsupdate.sh [hunk 2]
```diff
@@ -2,14 +2,25 @@
...
+    echo "Updating IPv4..." >> update.log
+    DATA=$(cat update.api | sed "s/%PASSWD%/$PASSWORD/g;s/%USER%/$USERNAME/g;s/%DNSID%/$DNSIDv4/g;s/%NEWIP%/$NEWIPv4/g")
+    curl  -s -X POST -d "$DATA" "$APIHOST" --header "Content-Type:text/xml" >> update.log
+fi
+
+if [ ! "$OLDIPv6" == "$NEWIPv6" ]; then
+    echo $NEWIP > old.ipv6
```
Comment (gehaxelt) [inline dnsupdate.sh]: Same as above, but only with ```$NEWIP``` -> ```$NEWIPv6```

**Comment by gehaxelt** [comment]
Another note:

I think that your improved logging by using ```>>``` instead of ```>``` may cause the logfiles to grow rapidly and thus consuming a lot of space. 

Maybe adding a switch ```$CLEARLOG=1``` which can be set to either ```1``` or ```0``` to check if the logfile should be cleared  or not?

Something like in the beginning?:

```
if [ $CLEARLOG -eq 1 ]; then
   rm update.log
fi
```

What do you think?

File: dnsupdate.sh [hunk 3]
```diff
@@ -2,14 +2,25 @@
...
-DNSID="" #DNS Entry ID
+DNSIDv4="" #DNS Entry ID
+DNSIDv6="" #DNS Entry ID
 APIHOST="https://api.domrobot.com/xmlrpc/"
-OLDIP=$(cat old.ip)
-NEWIP=$(curl -s curlmyip.com)
+OLDIPv4=$(cat old.ipv4)
+OLDIPv6=$(cat old.ipv6)
+NEWIPv4=$(curl -s curlmyip.com)
+NEWIPv6=$(curl -s curlmyip6.com)
```
Comment (gehaxelt) [inline dnsupdate.sh]: What about moving the ```NEWIPv6``` assignment down to line 19? If only IPv4 is used, we do not need to send the second request. Same idea for the ```OLDIPv6``` line.

File: dnsupdate.sh (hunk 1 above)
Comment (joker234) [inline dnsupdate.sh]: fixed

File: dnsupdate.sh (hunk 2 above)
Comment (joker234) [inline dnsupdate.sh]: fixed

File: dnsupdate.sh (hunk 3 above)
Comment (joker234) [inline dnsupdate.sh]: What exactly do you mean? How do I know that there is no IPv6 before `curl -s curlmyip6.com`?

**Comment by joker234** [comment]
Yes, you may be right with the log. In my opinion the log doesn't grow that much, because the log will be only written if the IP of your device changes (all `> update.log` are in the if terms).

**Comment by gehaxelt** [comment]
Hi @joker234,
if you could implement ```$CLEARLOG``` that would be great!

File: dnsupdate.sh (hunk 3 above)
Comment (gehaxelt) [inline dnsupdate.sh]: I was thinking about the following structure:

```
$OLDIPv4=$(cat old.ipv4)
$NEWIPv4=$(curl -s ...)

if [ $OLDIPv4 ... ]
//Update IPv4 here
fi

$OLDIPv6=$(cat old.ipv6)
$NEWIPv6=$(curl -s ...)

if [ $OLDIP6 ... ]
//Update IPv6 here
fi
```

**Comment by joker234** [comment]
done and commented

**Comment by gehaxelt** [comment]
Looks good!

Thanks :)

**Pull request merged**